### PR TITLE
MITO-112: integration test with ism-multisig

### PR DESCRIPTION
**Pull request by byeongsu-hong**
* change type of domain (u64 -> u32)
* test validator set that supports random / derive mode

### CODE REVIEW

File: integration-test/tests/validator.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+fn pub_to_addr(pub_key: Binary, prefix: &str) -> String {
+    let sha_hash = sha256_digest(pub_key);
+    let rip_hash = ripemd160_digest(sha_hash);
+
+    let addr = hpl_interface::types::bech32_encode(prefix, &rip_hash).unwrap();
+
+    addr.to_string()
+}
```
Comment (hashableric) [inline integration-test/tests/validator.rs:35]: ask; why don't we move those veirifcation functions to general utils?

File: packages/interface/src/types/bech32.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        if check == 0 {
+            return bech32_encode(hrp, &raw_addr[12..]);
+        }
+    }
+
+    let enc_addr = bech32::encode(hrp, raw_addr.to_base32(), bech32::Variant::Bech32)
+        .map_err(|e| StdError::generic_err(format!("invalid bech32 address. err: {e}")))?;
+
+    Ok(Addr::unchecked(enc_addr))
+}
```
Comment (hashableric) [inline packages/interface/src/types/bech32.rs:45]: ask; this too. move general libs

**Review by hashableric — APPROVED** [review]
not include important claim.

File: integration-test/tests/validator.rs (hunk 1 above)
Comment (byeongsu-hong) [reply integration-test/tests/validator.rs:35]: This uses hash libraries that may not be used by some contracts

File: packages/interface/src/types/bech32.rs (hunk 2 above)
Comment (byeongsu-hong) [reply packages/interface/src/types/bech32.rs:45]: hmm is it already placed on general libs?

**Comment by byeongsu-hong** [comment]
Merged - Please submit additional issues if you found something

**Pull request merged**